RUFAS/routines/feed_storage/storage.py

```diff
@@ -215,65 +214,87 @@ def record_stored_crops(self, gaseous_dry_matter_loss: float) -> None:
         but the structure of the loss equation remains the same.
 
         """
-        dry_matter_fraction = crop.dry_matter_percentage / 100
-        average_temperature = current_conditions.mean_air_temperature
+        dry_matter_fraction = crop.dry_matter_percentage * GeneralConstants.PERCENTAGE_TO_FRACTION
 
         is_alfalfa = crop.category is CropCategory.ALFALFA
-        lower_temp_limit = (
-            ALFALFA_GASEOUS_LOSS_LOWER_TEMP_LIMIT if is_alfalfa else NON_ALFALFA_GASEOUS_LOSS_LOWER_TEMP_LIMIT
-        )
-        upper_temp_limit = (
-            ALFALFA_GASEOUS_LOSS_UPPER_TEMP_LIMIT if is_alfalfa else NON_ALFALFA_GASEOUS_LOSS_UPPER_TEMP_LIMIT
-        )
-        lower_dry_matter_limit = (
-            ALFALFA_GASEOUS_LOSS_LOWER_DRY_MATTER_LIMIT
-            if is_alfalfa
-            else NON_ALFALFA_GASEOUS_LOSS_LOWER_DRY_MATTER_LIMIT
-        )
-        upper_dry_matter_limit = (
-            ALFALFA_GASEOUS_LOSS_UPPER_DRY_MATTER_LIMIT
-            if is_alfalfa
-            else NON_ALFALFA_GASEOUS_LOSS_UPPER_DRY_MATTER_LIMIT
-        )
+        constants = ALFALFA_FERMENTATION_CONSTANTS if is_alfalfa else NON_ALFALFA_FERMENTATION_CONSTANTS
+        lower_temp_limit = constants["lower_temp_limit"]
+        upper_temp_limit = constants["upper_temp_limit"]
+        lower_dry_matter_limit = constants["lower_dry_matter_limit"]
+        upper_dry_matter_limit = constants["upper_dry_matter_limit"]
+        loss_coefficient = constants["loss_coefficient"]
+        base_loss_fraction = constants["base_loss_fraction"]
+
+        dry_matter_loss_fraction = 0.0
+
+        for day in conditions:
+            outside_temp_range = not lower_temp_limit <= day.mean_air_temperature <= upper_temp_limit
+            outside_dry_fraction_range = not lower_dry_matter_limit <= dry_matter_fraction <= upper_dry_matter_limit
+            if outside_temp_range or outside_dry_fraction_range:
+                continue
+
+            fraction_lost = base_loss_fraction - loss_coefficient * (dry_matter_fraction - lower_dry_matter_limit)
+            dry_matter_loss_fraction += fraction_lost
+            dry_matter_fraction -= fraction_lost
 
-        if (not lower_temp_limit <= average_temperature <= upper_temp_limit) or (
-            not lower_dry_matter_limit <= crop.dry_matter_percentage <= upper_dry_matter_limit
-        ):
-            return 0.0
+        return crop.dry_matter_mass * dry_matter_loss_fraction
 
-        if is_alfalfa:
-            dry_matter_loss_fraction = 0.0156 - 0.0364 * (dry_matter_fraction - 0.20)
-        else:
-            dry_matter_loss_fraction = 0.00864 - 0.0193 * (dry_matter_fraction - 0.15)
+    def _get_conditions(
+        self, last_degradations_time: Time, current_time: Time, weather: Weather
+    ) -> list[CurrentDayConditions]:
+        """
+        Gets a series of weather conditions for a time period which will have weather conditions calculated for it.
```

Comment:
This description is a little confusing for me. I made a suggestion but not sure if it's capturing what's happening still.
```suggestion
        Gets a series of weather conditions for a time period.
```

Reply:
Good catch, I don't even know what I meant.